upgrades/steps_24.go

```diff
@@ -36,3 +46,47 @@ func stateStepsFor24() []Step {
 		},
 	}
 }
+
+// stepsFor24 returns upgrade steps for Juju 2.4.
+func stepsFor24() []Step {
+	return []Step{
+		&upgradeStep{
+			description: "Install the service file in Standard location '/lib/systemd'",
+			targets:     []Target{AllMachines},
+			run:         installServiceFile,
+		},
+	}
+}
+
+// install the service files in Standard location - '/lib/systemd/juju-init/.
+func installServiceFile(context Context) error {
+	hostSeries, err := series.HostSeries()
+	if err == nil {
+		initName, err := service.VersionInitSystem(hostSeries)
+		if err != nil {
+			logger.Errorf("Unsuccessfull installing the servie files in /lib/systemd/...")
+			return err
+		} else {
+			if initName == service.InitSystemSystemd {
+				oldDataDir := context.AgentConfig().DataDir()
+				oldInitDataDir := filepath.Join(oldDataDir, "init")
+
+				var sysdManager service.SystemdServiceManager
+				err = sysdManager.WriteServiceFile()
```

Comment:
As discussed in IRC, you'll need to set sysdManager before calling this - otherwise it's nil here and you'll get a nil pointer panic.